arch/avr/src/avr/up_createstack.c

```diff
@@ -86,10 +86,6 @@
 
 int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 {
-  /* Add the size of the TLS information structure */
-
-  stack_size += sizeof(struct tls_info_s);
-
 #ifdef CONFIG_TLS_ALIGNED
   /* The allocated stack size must not exceed the maximum possible for the
    * TLS feature.
```

Comment:
are the lines below (checking stack_size against TLS_MAXSTACK) still valid? I understand that stack_size no longer includes TLS size, right?

Reply:
Yes, TLS_MAXSTACK is still valid because it's key invariant of the aligned TLS. The real difference is that the stack size is always equals to caller supplied value without adjustion(+ sizeof(struct tls_info_s)).

Reply:
The problem is resovled by moving the adjustiont to common place:
https://github.com/apache/incubator-nuttx/pull/3517/files#diff-3e8be9133fa805536122dc95c4303786822b38382e44e053b399e5a43824ae9fR123-R125
https://github.com/apache/incubator-nuttx/pull/3517/files#diff-06984fa681053d61fe6d98716085fb7d1e9099deecbb332fa5da35d05b28ee1fR166-R169
So the behaviour is same as before.